contrib/new_post.py

```diff
@@ -0,0 +1,183 @@
+        words.pop(1)
+
+    # If the first word is a component with square brackets or colon, drop first word.
+    words[0] = words[0].lower()  # lowercase the first word in place
+    prefix = words[0].replace(':[]', '')  # prefix is first word stripped of :[] chars
+    if words[0] != prefix and is_a_component(prefix):
+        words.pop(0)
+
+    # Capitalize the first word
+    words[0] = words[0].capitalize()
+
+    # Return enclosed in double quotes after joining words and removing any double quotes.
+    title = " ".join(words).replace("\"", "")
+    return f"\"{title}\""
+
+def valid_components(labels):
+    return [l for l in labels if l in COMPONENTS]
+
+def invalid_components(labels):
+    return [l for l in labels if l not in COMPONENTS]
+
+def get_nonempty_components(labels):
+    # Parses the GitHub labels, and requires user input if no valid components were found
+    components = valid_components(labels)
+    if components:
+        return components
+
+    print(f"No label assigned to the PR yet; you will need to add one or more (comma-separated) manually from {COMPONENTS}")
+    while True:
+        components_input = input().replace("\'\"", '').split(',')
+        components_input = [c.strip().lower() for c in components_input]
+
+        if invalid_components(components_input):
+            print(f"Components {invalid_components(components_input)} are invalid, please try again")
+            continue
+
+        return components_input
+
+def create_post_file(fname, pr, date, host):
+    title = parse_title(pr.title)
+    labels = [l.lower() for l in pr.labels]
+    components = get_nonempty_components(labels)
+
+    print(f"GitHub PR title: \"{pr.title}\"")
+    print(f"Parsed PR title: {title}")
+    print(f"GitHub PR labels: \"{labels}\"")
+    print(f"Parsed PR labels: \"{components}\"")
+
+    with open(fname, 'w') as f:
+        f.write('---\n')
+        f.write('layout: pr\n')
+        f.write(f"date: {date}\n")
+        f.write(f"title: {title}\n")
+        f.write(f"pr: {pr.number}\n")
+        f.write(f"authors: [{pr.user}]\n")
+        f.write(f"components: {json.dumps(components)}\n")  # Use json.dumps() for double quotes
+        f.write(f"host: {host}\n")
+        f.write("status: upcoming\n")
+        f.write("commit:\n")
+        f.write("---\n\n")
+        f.write("_Notes and questions to follow soon!_\n\n")
+        f.write("<!-- TODO: Before meeting, add notes and questions\n")
+        f.write("## Notes\n\n\n\n")
+        f.write("## Questions\n\n")
+        f.write("1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?\n")
+        f.write("-->\n\n\n")
+        f.write("<!-- TODO: After meeting, uncomment and add meeting log between the irc tags\n")
+        f.write("## Meeting Log\n\n")
+        f.write("{% irc %}\n")
+        f.write("{% endirc %}\n")
+        f.write("-->\n")
```

Comment:
Alternatively (if you don't like the visual blank lines, can still use `\n` with this approach):

```suggestion
   body = textwrap.dedent(
        f"""
        ---
        layout: pr
        date: {date}
        title: {title}
        pr: {pr.number}
        authors: [{pr.user}]
        components: {json.dumps(components)}  # Use json.dumps() for double quotes
        host: {host}
        status: upcoming
        commit:
        ---

        _Notes and questions to follow soon!_

        <!-- TODO: Before meeting, add notes and questions
        ## Notes



        ## Questions

        1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
        -->


        <!-- TODO: After meeting, uncomment and add meeting log between the irc tags
        ## Meeting Log

        {{% irc %}}
        {{% endirc %}}
        -->
        """)

    with open(fname, 'w') as f:
        f.writelines(body)
```

Reply:
I'm going to leave this for now. Longer term I think the best thing to do here would be to have a template post file and use a templating engine like [jinja](https://jinja.palletsprojects.com/en/3.1.x/) to generate the new file.